src/main/java/org/jfrog/hudson/pipeline/common/executors/ExtractorJarExecutor.java

```diff
@@ -0,0 +1,72 @@
+package org.jfrog.hudson.pipeline.common.executors;
+
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.util.ArgumentListBuilder;
+import org.apache.commons.lang.StringUtils;
+import org.jfrog.build.api.BuildInfoFields;
+import org.jfrog.hudson.pipeline.common.Utils;
+import org.jfrog.hudson.pipeline.common.types.buildInfo.BuildInfo;
+import org.jfrog.hudson.util.ExtractorUtils;
+import org.jfrog.hudson.util.PluginDependencyHelper;
+
+import java.util.Objects;
+
+/**
+ * Created by Bar Belity on 08/07/2020.
+ */
+public abstract class ExtractorJarExecutor implements Executor {
+
+    TaskListener listener;
+    BuildInfo buildInfo;
+    Launcher launcher;
+    String javaArgs;
+    FilePath ws;
+    String path;
+    String module;
+    EnvVars env;
+    Run build;
+
+    public ExtractorJarExecutor(BuildInfo buildInfo, Launcher launcher, String javaArgs, FilePath ws, String path, String module, EnvVars env, TaskListener listener, Run build) {
```

Comment:
ExtractorJarExecutor --> BuildInfoProcessRunner

Let's also add some description for this class.